sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -78,53 +115,100 @@ private void canvasRestApiCall(HttpServletRequest request,
 		String queryString = request.getQueryString();
 		String pathInfo = request.getPathInfo();
 		String url=null;
-		if(pathInfo!=null&&(pathInfo.contains("courses")||pathInfo.contains("terms")||pathInfo.contains("crosslist"))) {
-		if(queryString!=null) {
-		url= canvasURL+pathInfo+"?"+queryString;
-		 }else {
-			url=canvasURL+pathInfo;
-		}
-		String sessionId = request.getSession().getId();
-		StringBuilder loggingApiWithSessionInfo =new StringBuilder();
-		loggingApiWithSessionInfo.append("Canvas API request with Session Id:  \"");
-		loggingApiWithSessionInfo.append(sessionId);
-		loggingApiWithSessionInfo.append("\" for URL:  \"");
-		loggingApiWithSessionInfo.append(url);
-		loggingApiWithSessionInfo.append("\"");
-		M_log.info(loggingApiWithSessionInfo);
-		HttpUriRequest clientRequest = null;
-		if(request.getMethod().equals(GET)) {
-			clientRequest = new HttpGet(url);
-		}else if (request.getMethod().equals(POST)) {
-			clientRequest = new HttpPost(url);
-		}else if(request.getMethod().equals(PUT)) {
-			clientRequest=new HttpPut(url);
-		}
-		HttpClient client = new DefaultHttpClient();
-		final ArrayList<NameValuePair> nameValues = new ArrayList<NameValuePair>();
-		nameValues.add(new BasicNameValuePair("Authorization", "Bearer"+ " " +canvasToken));
-		nameValues.add(new BasicNameValuePair("content-type", "application/json"));
-		for (final NameValuePair h : nameValues)
-		{
-		    clientRequest.addHeader(h.getName(), h.getValue());
-		}
-		BufferedReader rd = null;
-		try {
-			 rd = new BufferedReader(new InputStreamReader(client.execute(clientRequest).getEntity().getContent()));
-		} catch (IOException e) {
-			M_log.error("Canvas API call did not happen",e);
-		}
-		String line = "";
-		StringBuilder sb = new StringBuilder();
-		while ((line = rd.readLine()) != null) {
-			sb.append(line);
+		if(isAllowedApiRequest(request)) {
+			if(queryString!=null) {
```

Comment:
Suggest making the if and else clauses both smaller separate methods to make the logic and functionality clearer, easier to read, and easier to test.